### PR TITLE
fix(ivy): change detection strategy not being passed to compiler

### PR DESCRIPTION
**Pull request by crisbeto**
Fixes the defined change detection strategy not being passed to the compiler when a component is being compiled.

### CODE REVIEW

File: packages/compiler/src/compiler_facade_interface.ts [hunk 1]
```diff
@@ -6,6 +6,7 @@
  * found in the LICENSE file at https://angular.io/license
  */
 
+import {ChangeDetectionStrategy} from './core';
```
Comment (AndrewKushnir) [inline packages/compiler/src/compiler_facade_interface.ts]: It looks like `ChangeDetectionStrategy` should be consistent with the `ViewEncapsulation` here, i.e. it should be defined as `number` in `R3ComponentMetadataFacade`.

**Review by mhevery — APPROVED** [review]

File: packages/core/src/render3/jit/compiler_facade_interface.ts [hunk 2]
```diff
@@ -6,6 +6,7 @@
  * found in the LICENSE file at https://angular.io/license
  */
 
+import {ChangeDetectionStrategy} from '../../core';
```
Comment (AndrewKushnir) [inline packages/core/src/render3/jit/compiler_facade_interface.ts]: Please make it consistent with `ViewEncapsulation` in this file as well.

**Review by AndrewKushnir — CHANGES_REQUESTED** [review]
@crisbeto looks great! I just noticed a few inconsistencies between `ViewEncapsulation` and `ChangeDetection` handling (see my comments). Also it looks like we need to add `ChangeDetection` handling in `packages/compiler-cli/src/ngtsc/annotations/src/component.ts` (to align it with `ViewEncapsulation`).

**Comment by crisbeto** [comment]
Thank you for the feedback @AndrewKushnir, it's addressed now. Can you take another look?

**Review by kara — APPROVED** [review]
LGTM

**Comment by kara** [comment]
[presubmit](http://test/OCL:226330285:BASE:226330300:1545318238295:e141ea3c)

**Review by AndrewKushnir — APPROVED** [review]
LGTM, thanks for addressing my comments!

**Pull request closed without merging**